File: python/clock.py
from typing import Self


class Clock:

    def __init__(self, hour: int, minute: int):
        self.hour = hour
        self.minute = minute
        self._simplify()

    def __repr__(self) -> str:
        return f'Clock({self.hour}, {self.minute})'

    def __str__(self) -> str:
        return f'{self.hour:02}:{self.minute:02}'

    def __eq__(self, other: Self) -> bool:
        return self.hour == other.hour and self.minute == other.minute

    def __add__(self, minutes: int):
        return Clock(self.hour, self.minute + minutes)

    def __sub__(self, minutes: int):
        return Clock(self.hour, self.minute - minutes)

    def _simplify(self):
        self.hour += self.minute // 60
        self.minute %= 60
        self.hour %= 24

File: go/clock.go
package clock

import (
	"fmt"
)

// Clock ...
type Clock struct {
	hour   int
	minute int
}

// New ...
func New(hour, minute int) Clock {
	for minute < 0 {
		minute += 60
		hour--
	}

	for hour < 0 {
		hour += 24
	}

	hour += minute / 60

	return Clock{
		hour:   hour % 24,
		minute: minute % 60,
	}
}

// String ...
func (c Clock) String() string {
	return fmt.Sprintf("%02d:%02d", c.hour, c.minute)
}

// Add ...
func (c Clock) Add(minutes int) Clock {
	return New(c.hour, c.minute+minutes)
}

// Subtract ...
func (c Clock) Subtract(minutes int) Clock {
	return New(c.hour, c.minute-minutes)
}
